forms/gridfield/GridFieldExportButton.php

```diff
@@ -115,12 +115,14 @@ protected function getExportColumnsForGridField(GridField $gridField) {
 	 * Generate export fields for CSV.
 	 *
 	 * @param GridField $gridField
-	 * @return array
+	 * @return string
 	 */
 	public function generateExportFileData($gridField) {
 		$separator = $this->csvSeparator;
 		$csvColumns = $this->getExportColumnsForGridField($gridField);
-		$fileData = '';
+
+		$stream = fopen('php://memory', 'w');
```

Comment:
My suggestion is to put this entire structure into a try {} finally {}, so you can safely nest output buffering and cleanup in finally.

If you are writing directly to a memory stream, you shouldn't need to use the output buffering feature, should you? As I understand it, ob_start(); affects the `php://output stream`.